### PR TITLE
ref: freight 2022 refresh

### PR DESCRIPTION
**Pull request by joshuarli**
Big refresh, hot shower, whatever you wanna call it.

- Local dev requires postgres 9.6, which is not possible to install via brew these days. I've modified the old formula to be a bit leaner and it can be installed with `make mac-install-postgres-96`.
- Various core and dev requirements upgrades (except flask, redis, raven -> sentry_sdk) as well as Python 3.8.1 -> 3.8.12. 
- Switched to pre-commit and ran it on everything.
- Cleaned up the Dockerfile.

### CODE REVIEW

**Review by dbennett-sentry — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -1,76 +1,13 @@
...
-    && ln -s pip3 pip \
-    && ls -Fla /usr/local/bin/p* \
-    && which python  && python -V \
-    && which python2 && python2 -V \
-    && which python3 && python3 -V \
-    && which pip     && pip -V \
-    && which pip2    && pip2 -V \
-    && which pip3    && pip3 -V
-
-RUN pip2 install virtualenv
```
Comment (joshuarli) [inline Dockerfile:67]: Alright so, good learning from here is that this is used by https://github.com/getsentry/getsentry-deploy/blob/master/getsentry/fab.

So, getsentry-deploy needs porting to Python 3.8. I tried a few hotfixes but there's a harder issue with fabric's paramiko: https://freight.getsentry.net/deploys/getsentry/production/13207

```
+ exec /workspace/freight-workspace-6279ab38cb1311eca39f0242ac130003/getsentry-deploy/getsentry/.freight-env/bin/fab -f /workspace/freight-workspace-6279ab38cb1311eca39f0242ac130003/getsentry-deploy/getsentry/fabfile.py --colorize-errors -a -i /tmp/tmpq10tgdzw -R production deploy:branch_name=e5d5e9c546da9a25ee93cde9383b167225092426,prev_sha=e073efa431416cf46673f22b96cfdbfef2e22d71[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L43)
/workspace/freight-workspace-6279ab38cb1311eca39f0242ac130003/getsentry-deploy/getsentry/.freight-env/lib/python3.8/site-packages/paramiko/transport.py:236: CryptographyDeprecationWarning: Blowfish has been deprecated[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L44)
  "class": algorithms.Blowfish,[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L45)
[garbage.i.getsentry.net] Executing task 'deploy'[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L46)
/usr/local/lib/python3.8/getpass.py:91: GetPassWarning: Can not control echo on the terminal.[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L47)
  passwd = fallback_getpass(prompt, stream)[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L48)
Warning: Password input may be echoed.[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L49)
[garbage.i.getsentry.net] Login password for 'build': Traceback (most recent call last):[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L50)
  File "/workspace/freight-workspace-6279ab38cb1311eca39f0242ac130003/getsentry-deploy/getsentry/.freight-env/lib/python3.8/site-packages/fabric/network.py", line 478, in connect[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L51)
    client.connect(**kwargs)[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L52)
  File "/workspace/freight-workspace-6279ab38cb1311eca39f0242ac130003/getsentry-deploy/getsentry/.freight-env/lib/python3.8/site-packages/paramiko/client.py", line 435, in connect[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L53)
    self._auth([12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L54)
  File "/workspace/freight-workspace-6279ab38cb1311eca39f0242ac130003/getsentry-deploy/getsentry/.freight-env/lib/python3.8/site-packages/paramiko/client.py", line 766, in _auth[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L55)
    raise saved_exception[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L56)
  File "/workspace/freight-workspace-6279ab38cb1311eca39f0242ac130003/getsentry-deploy/getsentry/.freight-env/lib/python3.8/site-packages/paramiko/client.py", line 678, in _auth[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L57)
    key = self._key_from_filepath([12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L58)
  File "/workspace/freight-workspace-6279ab38cb1311eca39f0242ac130003/getsentry-deploy/getsentry/.freight-env/lib/python3.8/site-packages/paramiko/client.py", line 588, in _key_from_filepath[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L59)
    key = klass.from_private_key_file(key_path, password)[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L60)
  File "/workspace/freight-workspace-6279ab38cb1311eca39f0242ac130003/getsentry-deploy/getsentry/.freight-env/lib/python3.8/site-packages/paramiko/pkey.py", line 249, in from_private_key_file[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L61)
    key = cls(filename=filename, password=password)[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L62)
  File "/workspace/freight-workspace-6279ab38cb1311eca39f0242ac130003/getsentry-deploy/getsentry/.freight-env/lib/python3.8/site-packages/paramiko/ed25519key.py", line 58, in __init__[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L63)
    pkformat, data = self._read_private_key("OPENSSH", f)[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L64)
  File "/workspace/freight-workspace-6279ab38cb1311eca39f0242ac130003/getsentry-deploy/getsentry/.freight-env/lib/python3.8/site-packages/paramiko/pkey.py", line 354, in _read_private_key[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L65)
    raise SSHException([12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L66)
paramiko.ssh_exception.SSHException: encountered RSA key, expected OPENSSH key[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L67)
[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L68)
During handling of the above exception, another exception occurred:[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L69)
[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L70)
Traceback (most recent call last):[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L71)
  File "/usr/local/lib/python3.8/getpass.py", line 69, in unix_getpass[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L72)
    old = termios.tcgetattr(fd)     # a copy to save[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L73)
termios.error: (25, 'Inappropriate ioctl for device')[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L74)
[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L75)
During handling of the above exception, another exception occurred:[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L76)
[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L77)
Traceback (most recent call last):[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L78)
  File "/workspace/freight-workspace-6279ab38cb1311eca39f0242ac130003/getsentry-deploy/getsentry/.freight-env/lib/python3.8/site-packages/fabric/main.py", line 758, in main[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L79)
    execute([12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L80)
  File "/workspace/freight-workspace-6279ab38cb1311eca39f0242ac130003/getsentry-deploy/getsentry/.freight-env/lib/python3.8/site-packages/fabric/tasks.py", line 385, in execute[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L81)
    results[host] = _execute([12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L82)
  File "/workspace/freight-workspace-6279ab38cb1311eca39f0242ac130003/getsentry-deploy/getsentry/.freight-env/lib/python3.8/site-packages/fabric/tasks.py", line 277, in _execute[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L83)
    return task.run(*args, **kwargs)[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L84)
[12:01:20 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L85)
  File "/workspace/freight-workspace-6279ab38cb1311eca39f0242ac130003/getsentry-deploy/getsentry/.freight-env/lib/python3.8/site-packages/fabric/tasks.py", line 174, in run[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L86)
    return self.wrapped(*args, **kwargs)[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L87)
  File "/workspace/freight-workspace-6279ab38cb1311eca39f0242ac130003/getsentry-deploy/getsentry/.freight-env/lib/python3.8/site-packages/fabric/decorators.py", line 55, in inner_decorator[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L88)
    return func(*args, **kwargs)[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L89)
  File "/workspace/freight-workspace-6279ab38cb1311eca39f0242ac130003/getsentry-deploy/getsentry/fabfile.py", line 196, in deploy[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L90)
    clone_or_update_repo()[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L91)
  File "/workspace/freight-workspace-6279ab38cb1311eca39f0242ac130003/getsentry-deploy/getsentry/fabfile.py", line 106, in clone_or_update_repo[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L92)
    if not exists(os.path.join(path, 'master')):[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L93)
  File "/workspace/freight-workspace-6279ab38cb1311eca39f0242ac130003/getsentry-deploy/getsentry/.freight-env/lib/python3.8/site-packages/fabric/contrib/files.py", line 31, in exists[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L94)
    cmd = 'stat %s' % _expand_path(path)[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L95)
  File "/workspace/freight-workspace-6279ab38cb1311eca39f0242ac130003/getsentry-deploy/getsentry/.freight-env/lib/python3.8/site-packages/fabric/contrib/files.py", line 486, in _expand_path[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L96)
    return path if is_win() else '"$(echo %s)"' % path[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L97)
  File "/workspace/freight-workspace-6279ab38cb1311eca39f0242ac130003/getsentry-deploy/getsentry/.freight-env/lib/python3.8/site-packages/fabric/contrib/files.py", line 473, in is_win[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L98)
    return '"' in run('echo "Will you echo quotation marks"')[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L99)
  File "/workspace/freight-workspace-6279ab38cb1311eca39f0242ac130003/getsentry-deploy/getsentry/.freight-env/lib/python3.8/site-packages/fabric/network.py", line 692, in host_prompting_wrapper[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L100)
    return func(*args, **kwargs)[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L101)
  File "/workspace/freight-workspace-6279ab38cb1311eca39f0242ac130003/getsentry-deploy/getsentry/.freight-env/lib/python3.8/site-packages/fabric/operations.py", line 1092, in run[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L102)
    return _run_command([12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L103)
  File "/workspace/freight-workspace-6279ab38cb1311eca39f0242ac130003/getsentry-deploy/getsentry/.freight-env/lib/python3.8/site-packages/fabric/operations.py", line 935, in _run_command[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L104)
    channel=default_channel(), command=wrapped_command, pty=pty,[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L105)
  File "/workspace/freight-workspace-6279ab38cb1311eca39f0242ac130003/getsentry-deploy/getsentry/.freight-env/lib/python3.8/site-packages/fabric/state.py", line 435, in default_channel[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L106)
    chan = _open_session()[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L107)
  File "/workspace/freight-workspace-6279ab38cb1311eca39f0242ac130003/getsentry-deploy/getsentry/.freight-env/lib/python3.8/site-packages/fabric/state.py", line 416, in _open_session[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L108)
    transport = connections[env.host_string].get_transport()[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L109)
  File "/workspace/freight-workspace-6279ab38cb1311eca39f0242ac130003/getsentry-deploy/getsentry/.freight-env/lib/python3.8/site-packages/fabric/network.py", line 156, in __getitem__[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L110)
    self.connect(key)[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L111)
  File "/workspace/freight-workspace-6279ab38cb1311eca39f0242ac130003/getsentry-deploy/getsentry/.freight-env/lib/python3.8/site-packages/fabric/network.py", line 147, in connect[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L112)
    self[key] = connect([12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L113)
  File "/workspace/freight-workspace-6279ab38cb1311eca39f0242ac130003/getsentry-deploy/getsentry/.freight-env/lib/python3.8/site-packages/fabric/network.py", line 566, in connect[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L114)
    password = prompt_for_password(text)[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L115)
  File "/workspace/freight-workspace-6279ab38cb1311eca39f0242ac130003/getsentry-deploy/getsentry/.freight-env/lib/python3.8/site-packages/fabric/network.py", line 651, in prompt_for_password[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L116)
    new_password = _password_prompt(password_prompt, stream)[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L117)
  File "/workspace/freight-workspace-6279ab38cb1311eca39f0242ac130003/getsentry-deploy/getsentry/.freight-env/lib/python3.8/site-packages/fabric/network.py", line 623, in _password_prompt[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L118)
    return getpass.getpass(prompt, stream)[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L119)
  File "/usr/local/lib/python3.8/getpass.py", line 91, in unix_getpass[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L120)
    passwd = fallback_getpass(prompt, stream)[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L121)
  File "/usr/local/lib/python3.8/getpass.py", line 126, in fallback_getpass[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L122)
    return _raw_input(prompt, stream)[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L123)
  File "/usr/local/lib/python3.8/getpass.py", line 148, in _raw_input[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L124)
    raise EOFError[12:01:21 PM](https://freight.getsentry.net/deploys/getsentry/production/13207#L125)
EOFError
```

Comment (joshuarli) [reply Dockerfile:67]: Oh hmm, I could just convert SSH_PRIVATE_KEY from RSA/PEM format to OPENSSH. Started going down a whole rabbithole with trying to upgrade to fabric v2: https://github.com/getsentry/getsentry-deploy/pull/95

Comment (joshuarli) [reply Dockerfile:67]: So let's try staging deploying https://github.com/getsentry/getsentry/pull/7427 so I don't need to keep hotfixing getsentry-deploy, and with freight bumped back here.

Comment (joshuarli) [reply Dockerfile:67]: Perfect, repro'd: https://freight.getsentry.net/deploys/getsentry/staging/2611

Comment (joshuarli) [reply Dockerfile:67]: Awesome, I can't edit the databag.

```
$ knife data bag edit services/freight SSH_PRIVATE_KEY
ERROR: The object you are looking for could not be found
Response: <HTML><HEAD><TITLE>404 Not Found</TITLE></HEAD><BODY><H1>Not Found</H1>The requested document was not found on this server.<P><HR><ADDRESS>mochiweb+webmachine web server</ADDRESS></BODY></HTML>
```

I swear Chef was working for me before - I was able to read this recently, but now `show` fails the same way...

Comment (joshuarli) [reply Dockerfile:67]: Gonna try to just upgrade paramiko to a version accepted by fabric 1.15.0 but without that annoying change. If success then I'll f/u with a PR to at least remove the staging host from the fabfile + some cleanups.

**Pull request merged**